src/filter.rs

```diff
@@ -198,8 +198,9 @@ pub(crate) fn filter(method: FilterType, bpp: BytesPerPixel, previous: &[u8], cu
         }
         Avg => {
             for i in (bpp..len).rev() {
-                current[i] =
-                    current[i].wrapping_sub(current[i - bpp].wrapping_add(previous[i]) / 2);
+                current[i] = current[i].wrapping_sub(
+                    ((u16::from(current[i - bpp]) + u16::from(previous[i])) / 2) as u8,
+                );
```

Comment:
There is a `roundtrip` test some lines down. It seems to have missed this due to using very uniform input data which does not trigger the overflow bug. Could you add a second set of inputs that triger it, i.e. linear ascending sequence or something like that?

Reply:
I've added a test which the current master fails on.  I can combine it into one test if you think that's better.